File: week 3/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int main(void)
{
    const char *text = get_string("Text: ");
    int letters = 0;
    int words = 0;
    int sentences = 0;

    for (int i = 0; i < strlen(text) + 1; i++)
    {
        if (isalpha(text[i]))
        {
            letters++;
        }
        if (isspace(text[i]) || text[i] == '\0')
        {
            words++;
        }
        if (text[i] == '!' || text[i] == '?' || text[i] == '.')
        {
            sentences++;
        }
    }

    float l = (100 / (float)words) * letters;
    float s = (100 / (float)words) * sentences;

    float score = round(0.0588 * l - 0.296 * s - 15.8);

    if (score > 16)
    {
        printf("Grade 16+\n");
    }
    else if (score < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", (int)score);
    }
}

File: week 6/readability.py
from cs50 import get_string
import string

text = get_string("Text: ")
letters = 0
words = 1
sentences = 0


for i in range(len(text)):
    if text[i].isalpha():
        letters += 1
    if text[i].isspace():
        words += 1
    if text[i] == "!" or text[i] == "." or text[i] == "?":
        sentences += 1


l = (100 / words) * letters
s = (100 / words) * sentences

score = round(0.0588 * l - 0.296 * s - 15.8)

if score > 16:
    print("Grade 16+")

elif score < 1:
    print("Before Grade 1")
else:
    print(f"Grade {score}")
